Extended xsort test does not build in MSVC

xsort test fails to build with the following error

C:\Users\dahub\Source\Repos\xtensor\test\test_extended_xsort.cpp(110): error C2651: 'xt::xarray_container<xt::uvector<long,std::allocator<T>>,xt::layout_type::row_major,xt::svector<size_t,4,std::allocator<unsigned __int64>,true>,xt::xtensor_expression_tag> &': left of '::' must be a class, struct or union
          with
          [
              T=long
          ]

Looks to be an issue with type decay

--- a/test/test_extended_xsort.cpp
+++ b/test/test_extended_xsort.cpp
@@ -107,7 +107,7 @@ namespace xt
 
         // py_median = np.median(a)
         double py_median = 300.0;
-        EXPECT_EQ(static_cast<decltype(py_a)::value_type>(py_median), xt::median(py_a));
+        EXPECT_EQ(static_cast<std::decay<decltype(py_a)>::type::value_type>(py_median), xt::median(py_a));
     }
 
     /*py

